src/pure/stack_set.rs

```diff
@@ -799,6 +804,26 @@ impl StackSet<Xid> {
         Ok(())
     }
 
+    /// If a known client is floating, sink it.
+    /// Otherwise, record it as floating with its preferred screen position.
+    ///
+    /// # Errors
+    /// This method with return [Error::UnknownClient] if the given client is
+    /// not already managed in this stack_set.
+    ///
+    /// This method with return [Error::ClientIsNotVisible] if the given client is
+    /// not currently mapped to a screen. This is required to determine the correct
+    /// relative positioning for the floating client as is it is moved between
+    /// screens.
+    pub fn toggle_floating_state(&mut self, client: Xid, r: Rect) -> Result<Option<Rect>> {
+        Ok(if self.is_floating(&client) {
+            self.sink(&client)
+        } else {
+            self.float(client, r)?;
+            None
+        })
```

Comment:
Can this be bound to a variable and returned as `Ok(rect)` rather wrapping the whole if statement